dotnet/src/Functions/Functions.OpenApi/RestApiOperationRunner.cs

```diff
@@ -464,5 +477,60 @@ private Uri BuildsOperationUrl(RestApiOperation operation, IDictionary<string, o
         return new UriBuilder(url) { Query = operation.BuildQueryString(arguments) }.Uri;
     }
 
+    /// <summary>
+    /// Reads the HTTP content.
+    /// </summary>
+    /// <param name="requestMessage">The HTTP request message.</param>
+    /// <param name="responseMessage">The HTTP response message.</param>
+    /// <param name="mediaType">The media type of the content.</param>
+    /// <param name="cancellationToken">The cancellation token.</param>
+    /// <returns>The HTTP content.</returns>
+    private async Task<object?> ReadHttpContentAsync(HttpRequestMessage requestMessage, HttpResponseMessage responseMessage, string mediaType, CancellationToken cancellationToken)
+    {
+        object? content = null;
+
+        // Read content using the custom reader if provided.
+        if (this._httpResponseContentReader is not null)
+        {
+            content = await this._httpResponseContentReader.Invoke(new(requestMessage, responseMessage), cancellationToken).ConfigureAwait(false);
+        }
+
+        // If no custom reader is provided or the custom reader did not return any content, read the content using the default readers.
+        if (content is null)
+        {
+            // Obtain the content reader by media type (e.g., text/plain, application/json, image/jpg)
+            if (!s_contentReaderByContentType.TryGetValue(mediaType, out var reader))
```

Comment:
I don't think you are going to find the keys for `text` and `image` without ignoring the `subtype`

```suggestion
            if (!s_contentReaderByContentType.TryGetValue(mediaType, out var reader))
```

Reply:
Right, this line looks for a content reader by the full media type, such as `text/jpeg`. If no exact match is found, the code a few lines below, tries to resolve a reader by the primary content type, such as `text`.